processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -79,13 +78,42 @@ public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   @Override
   public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
-    throw new UnsupportedOperationException();
+    return factory.makeValueMatcher(
+        dimension, new Predicate<String>()
+        {
+          @Override
+          public boolean apply(String input)
+          {
+            // Assuming that a null/absent/empty dimension are equivalent from the druid perspective
+            return value.equals(Strings.nullToEmpty(fn.apply(Strings.emptyToNull(input))));
```

Comment:
[Value can't be null]( https://github.com/druid-io/druid/blob/61a5bb27847f329fe28b32c791bf3a4a9d315b45/processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java#L46)